### PR TITLE
[PCM-2187] - Hold other active sessions when handling a new session. 

### PR DESCRIPTION
**Pull request by zservies**

### CODE REVIEW

File: src/sessions/softphone-session-handler.ts [hunk 1]
```diff
@@ -88,6 +89,7 @@ export default class SoftphoneSessionHandler extends BaseSessionHandler {
         this can happen when LA > 1 and persistentConnection is ON
       */
       if (this.hasActiveSession() && !Object.values(this.conversations).find(c => c.session === this.activeSession)) {
+        console.warn('here are all the sessions: ', sessions);
```
Comment (hjon) [inline src/sessions/softphone-session-handler.ts:92]: This looks like a log that can be removed?

**Review by hjon — APPROVED** [review]

File: src/sessions/softphone-session-handler.ts (hunk 1 above)
Comment (zservies) [reply src/sessions/softphone-session-handler.ts:92]: oops

Comment (zservies) [reply src/sessions/softphone-session-handler.ts:92]: > This looks like a log that can be removed?

oops

File: src/sessions/softphone-session-handler.ts [hunk 2]
```diff
@@ -77,6 +76,8 @@ export default class SoftphoneSessionHandler extends BaseSessionHandler {
     /* lastly, look through our sessions */
     else {
       session = sessions.find(s => s.conversationId === update.id);
+      // Hold other sessions that were active
+      this.holdOtherSessions(session, sessions);
```
[the remark this answers is not in the document]
Comment (zservies) [reply src/sessions/softphone-session-handler.ts]: > 

@jensengar  You're right, I updated it to handle unholding sessions with LA>1.

File: src/sessions/softphone-session-handler.ts [hunk 3]
```diff
@@ -100,6 +101,17 @@ export default class SoftphoneSessionHandler extends BaseSessionHandler {
     this.handleSoftphoneConversationUpdate(update, participant, callState, session);
   }
 
+  holdOtherSessions(currentSession: IExtendedMediaSession): void {
+    const sessions = this.sessionManager.getAllActiveSessions();
+    const otherSessions = sessions.filter(session => session !== currentSession);
```
Comment (jensengar) [inline src/sessions/softphone-session-handler.ts:106]: Let's adjust this filter so it's only looking at softphone calls and also only looking for calls that are not already held.

**Review by jensengar — CHANGES_REQUESTED** [review]

File: src/sessions/softphone-session-handler.ts [hunk 4]
```diff
@@ -100,6 +102,18 @@ export default class SoftphoneSessionHandler extends BaseSessionHandler {
     this.handleSoftphoneConversationUpdate(update, participant, callState, session);
   }
 
+  holdOtherSessions(currentSession: IExtendedMediaSession): void {
+    const sessions = this.sessionManager.getAllActiveSessions();
+    // Hold only softphone sessions and sessions not currently held.
+    const otherSessions = sessions.filter(session => session.sessionType === SessionTypes.softphone && !session.pcParticipant?.calls[0].held && session !== currentSession);
```
Comment (zservies) [inline src/sessions/softphone-session-handler.ts:108]: I adjusted this to filter out anything not a softphone session and already held. I'm curious though, is there a better way to check if a session is currently held? I had to dig pretty deep on the session to find this @jensengar

**Review by jensengar — CHANGES_REQUESTED** [review]

**Review by maxwellmooney13 — APPROVED** [review]

**Pull request merged**